### PR TITLE
construct and assign portal and checkers consistently in ssh server example 9578

### PR DESCRIPTION
**Pull request by graingert**
https://twistedmatrix.com/trac/ticket/9578

## Contributor Checklist:

* [x] There is an associated ticket in Trac. Create a new one at https://twistedmatrix.com/trac/newticket
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests.

### CODE REVIEW

**Comment by cdunklau** [comment]
Looks better. Only thing I'd change is to accept the dependencies as args to the factory's `__init__` and just set them as attributes instead of initializing them inside. Then just create those things in the runner bit and pass them into `ExampleFactory(...)`.

File: docs/conch/examples/sshsimpleserver.py [hunk 1]
```diff
@@ -217,22 +217,20 @@ class ExampleFactory(factory.SSHFactory):
         b'ssh-connection': connection.SSHConnection
     }
 
+    def __init__(self):
```
Comment (markrwilliams) [inline docs/conch/examples/sshsimpleserver.py:228]: I agree with [@cdunklau's comment](https://github.com/twisted/twisted/pull/1096#issuecomment-457143556) - the constructor should accept these arguments.  That can be done in a separate PR, though.

**Review by markrwilliams — APPROVED** [review]
Thanks for the contribution! It's more idiomatic and it doesn't break the example, so :+1:

**Comment by glyph** [comment]
@markrwilliams - @graingert isn't a project member and can't merge this :).  Did you intend to land this change?

**Comment by markrwilliams** [comment]
@glyph Yes, but trunk was broken when I reviewed this!

**Comment by graingert** [comment]
@markrwilliams @glyph can you re-review?

**Comment by graingert** [comment]
@markrwilliams I can't see why this is failing CI - this is example code it's not actually run/tested?

**Comment by glyph** [comment]
The lint job is noisy, but it's this line:

```
./docs/conch/examples/sshsimpleserver.py:234:1: E305 expected 3 blank lines after class or function definition, found 2
```

**Comment by glyph** [comment]
The Windows failures on Azure *seem* spurious, but they make me suspicious since they're also Conch stuff.

**Pull request merged**